worksheet_D/oxo.py

```diff
@@ -0,0 +1,102 @@
+class OxoBoard:
+    def __init__(self):
+        self.gameboard = [[0, 0, 0], [0, 0, 0], [0, 0, 0]] #Initialises the gameboard list
+
+    def get_square(self, x, y):
+        return self.gameboard[x][y] #Returns the value of a square on the gameboard
+
+    def set_square(self, x, y, mark):
+        if self.gameboard[x][y] == 0:  #Checks if a square is empty, if so it's filled with mark, if not False is returned
+            self.gameboard[x][y] = mark
+            return True
+        else:
+            return False
+
+    def is_board_full(self):
+        for x in range(3):  #Returns a false boolean if any squares on the board are empty
+            for y in range(3):
+                if self.gameboard[x][y] == 0:
+                    return False
+        return True
+
+
+    def get_winner(self):
+        for i in range(3):
+            # Checks for Vertical matches, >0 checks that none of the squares are empty
+            if self.gameboard[0][i] == self.gameboard[1][i] == self.gameboard[2][i] > 0:
+                return self.gameboard[0][i]
+            # Checks for Horizontal matches
+            elif self.gameboard[i][0] == self.gameboard[i][1] == self.gameboard[i][2] > 0:
+                return self.gameboard[i][0]
+            # Checks for Diagonal matches
+            elif self.gameboard[0][2] == self.gameboard[1][1] == self.gameboard[2][0] > 0:
```

Comment:
The diagonal checks do not need to be inside the `for` loop -- you're checking the same thing 3 times